flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/context/SessionContext.java

```diff
@@ -275,4 +322,33 @@ private void resetSessionConfigurationToDefault(Configuration defaultConf) {
         }
         sessionConfiguration.addAll(defaultConf);
     }
+
+    private void buildClassLoaderAndUpdateDependencies(Collection<URL> newDependencies) {
```

Comment:
nit: the method name `buildClassLoader...` sounds like it returns a new classloader object, however it is not. Would be better to rename to `rebuildClassLoader..`.